test/subtype.jl

```diff
@@ -796,6 +800,11 @@ function test_intersection()
                    (@UnionAll X<:(@UnionAll T Tuple{T,T}) Ref{X}))
     @testintersect(Ref{@UnionAll T @UnionAll S Tuple{T,S}},
                    Ref{@UnionAll T Tuple{T,T}}, Bottom)
+
+    @testintersect(Tuple{T,T} where T<:Union{UpperTriangular, UnitUpperTriangular},
+                   Tuple{AbstractArray{T,N}, AbstractArray{T,N}} where N where T,
```

Comment:
💯 for `where`. Out of curiosity, would it be feasible (or desirable) to make the syntax `where N, T` rather than `where N where T`? Or has that ship sailed?